src/Adapter/Security/CommandHandler/BulkDeleteEmployeesSessionsHandler.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Security\CommandHandler;
+
+use EmployeeSession;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\SessionNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Command\BulkDeleteEmployeesSessionsCommand;
+use PrestaShop\PrestaShop\Core\Domain\Security\CommandHandler\BulkDeleteEmployeesSessionsHandlerInterface;
+
+/**
+ * Handles command that deletes employees sessions in bulk action.
+ *
+ * @internal
+ */
+final class BulkDeleteEmployeesSessionsHandler implements BulkDeleteEmployeesSessionsHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(BulkDeleteEmployeesSessionsCommand $command)
+    {
+        foreach ($command->getEmployeeSessionIds() as $sessionId) {
```

Comment:
Also it was recently decided (mostly by @eternoendless & agreed by others :smile: ), that we introduce a new layer inside handlers. Base idea is that handlers shouldn't contain the domain logic, but instead use domain services.
Not an ideal example yet, because its evolving, but you can see here https://github.com/PrestaShop/PrestaShop/pull/21117.

TLDR; we've introduced AbstractObjectModelRepository, which wraps basic persistence methods of object model.
So in case you need simple add/update/delete operation you can create e.g. EmployeeSessionRepository in Adapter & extend the AbstractObjectModelRepository.
That one should Provide,add, delete, update the object model wrapped in Domain exception.
Another case when your object model contains some domain logic, you can make some additional services for that which performs all of those actions and only then uses repository to persist them (like this one https://github.com/PrestaShop/PrestaShop/pull/21117/files#diff-af5330ba679fb4caf297b979a3b94126R44).

Reply:
Is there something written in the devdoc, I can't find anything related to it :thinking: 
I think we should write something about all code decisions because people like me are not following everything around the migration :sweat_smile:

Reply:
> Is there something written in the devdoc
nope, sorry :smile: 

You can check Adapter/Product/{domain} for examples, but basically you need to create some EmployeeSessionRepository->bulkDelete which could extend AbstractObjectModelRepository and inject it in your handler.

Reply:
:exploding_head:

Reply:
I really think this is something that should be documented somewhere, otherwise, I don't understand how people from the community or not follow the migration process can think about it.
This is something I will do when it will be in an ADR, or in the devdoc :) (It's not against you, but I don't want to spend time (this PR is not mandatory) trying to find files and hoping files are up to date :sweat_smile: )

Reply:
Ok, for people reading this, @zuk3975 already send me everything I wanted to know :heart_eyes_cat: